cpp/src/DataStorm/ForwarderManager.h

```diff
@@ -20,8 +21,38 @@ namespace DataStormI
         using Exception = std::function<void(std::exception_ptr)>;
 
         ForwarderManager(const Ice::ObjectAdapterPtr&, const std::string&);
-        Ice::ObjectPrx add(std::function<void(Ice::ByteSeq, Response, Exception, const Ice::Current&)>);
-        Ice::ObjectPrx add(std::function<void(Ice::ByteSeq, const Ice::Current&)>);
+
+        template<
+            typename Prx = Ice::ObjectPrx,
```

Comment:
I would remove the default for Prx. It's an internal API - we should update all callers.